public/app/features/plugins/datasource_srv.ts

```diff
@@ -32,6 +34,7 @@ export class DatasourceSrv implements DataSourceService {
   private settingsMapByName: Record<string, DataSourceInstanceSettings> = {};
   private settingsMapByUid: Record<string, DataSourceInstanceSettings> = {};
   private settingsMapById: Record<string, DataSourceInstanceSettings> = {};
+  private runtimeDataSources: Record<string, RuntimeDataSource> = {}; //
```

Comment:
Is this parallel map needed? Do we need to later figure out which ones are "runtime"?

Reply:
@aocenas yes, it is needed as we do reloads of data sources and then the runtime data sources would be lost